### PR TITLE
Add isAcceptedLicense to the test-suite

### PR DESCRIPTION
**Pull request by phadej**
Please include the following checklist in your PR:

* [ ] Patches conform to the [coding conventions](https://github.com/haskell/cabal/#conventions).
* [ ] Any changes that could be relevant to users have been recorded in the changelog.
* [ ] The documentation has been updated, if necessary.
* [ ] If the change is docs-only, `[ci skip]` is used to avoid triggering the build bots.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: Cabal/tests/UnitTests/Distribution/SPDX.hs [hunk 1]
```diff
@@ -55,6 +57,63 @@ reassoc (EAnd a b) = case reassoc a of
...
+    , simpleLicenseExpression BSD_4_Clause `EAnd` simpleLicenseExpression MIT
+    ]
+
+-- | A sketch of what Hackage could accept
+--
+-- * NONE is rejected
+--
+-- * "or later" syntax (+ postfix) is rejected
+--
+-- * "WITH exc" exceptions are rejected
```
Comment (ivan-m) [inline Cabal/tests/UnitTests/Distribution/SPDX.hs]: Won't this reject [HaXml](http://hackage.haskell.org/package/HaXml-1.25.4/src/COPYRIGHT)?

Comment (hvr) [reply Cabal/tests/UnitTests/Distribution/SPDX.hs]: @ivan-m what would be the SPDX expression for accurately describing HaXml's licence?

Comment (gbaz) [reply Cabal/tests/UnitTests/Distribution/SPDX.hs]: this thread ends up  listing more standard licenses that seem to take the place of LGPL+linking exception: https://softwareengineering.stackexchange.com/questions/179084/is-there-a-modified-lgpl-license-that-allows-static-linking

**Comment by 23Skidoo** [comment]
Only touches the test suite and is green, merging.

**Comment by 23Skidoo** [comment]
Merged, thanks!

**Pull request merged**